addons/survey/static/src/question_page/question_page_one2many_field.js

```diff
@@ -12,6 +14,87 @@ class QuestionPageOneToManyField extends X2ManyField {
+        // immediately during question creation.
+        // Preparing everything in order to override `this._openRecord` below.
+        const { saveRecord: superSaveRecord, updateRecord: superUpdateRecord } = useX2ManyCrud(
+            () => this.list,
+            this.isMany2Many
+        );
+
+        const self = this;
+        const saveRecord = async (record) => {
+            const saveResponse = await superSaveRecord(record);
+            try {
+                await self.props.record.save({stayInEdition: true, throwOnError: true});
+            } catch (error) {
+                return self.handleSurveySaveError(error, record);
+            }
+            return saveResponse;
+        };
+
+        const updateRecord = async (record) => {
+            const updateResponse = await superUpdateRecord(record);
+            try {
+                await self.props.record.save({stayInEdition: true, throwOnError: true});
+            } catch (error) {
+                return self.handleSurveySaveError(error);
+            }
+            return updateResponse;
+        };
+
+        const openRecord = useOpenX2ManyRecord({
+            resModel: this.list.resModel,
+            activeField: this.activeField,
+            activeActions: this.activeActions,
+            getList: () => this.list,
+            saveRecord,
+            updateRecord,
+        });
+        this._openRecord = async (params) => {
+            if (!await self.props.record.save({stayInEdition: true})) {
+                return;
+            }
+            const activeElement = document.activeElement;
+            if (params.record) {
+                // Trigger loading for values that may have changed when reordering/deleting
+                // rows (allowed_triggering_question_ids, is_conditional, is_placed_before_trigger).
+                await params.record.load();
+            }
+            openRecord({
+                ...params,
+                onClose: () => {
+                    if (activeElement) {
+                        activeElement.focus();
+                    }
+                },
+            });
+        };
+    }
+
+    /**
+     * As it is convenient to keep the questions modal open, two things should
+     * be cared for in case of error while saving the form after adding or
+     * updating a question:
+     *   * Remove erroneous question row added
+     *   * Replace default error modal with a notification
+     */
+    async handleSurveySaveError(error, recordToDelete) {
+        error.event.preventDefault();
+        if (recordToDelete) {
+            const listRecord = this.list.records.find(r => r.__bm_handle__ === recordToDelete.__bm_handle__);
+            await this.list.delete(listRecord.id);
+        }
+        this.notificationService.add(
+            error.message.data.message.replace(_lt('The operation cannot be completed: '), ''), {
```

Comment:
Mmh still not a fan.

If someday the sentence is reworked and we put a "." instead of a ":", there is no way we will modify your code at the same time ^^

Does not seem very important and could easily break, I would remove that post-processing, unless it's very critical somehow?